docs/rendering_system.rst

```diff
@@ -0,0 +1,241 @@
+titles are given a self-link.
+
+
+3.4.5 ``PEPFooter`` transform
+*****************************
+
+This firstly builds a map of file modification times from a single git call, as
+a speed-up. This will return incorrect results on a shallow check-out of the
+repository, as is the default on continuous integration systems.
+
+We then attempt to remove any empty references sections, and append metadata in
+the footer (source link and last modified timestamp).
+
+
+3.5 Prepare for writing
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+``pep_html_builder.FileBuilder.prepare_writing`` initialises the bare miniumum
+of the Docutils writer and the settings for writing documents.
+This provides a significant speed-up over the base Sphinx implementation as most
+of the data automatically initialised was unused.
+
+
+3.6 Translate Docutils to HTML
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+``PEPTranslator`` overrides paragraph and reference logic to replicate
+processing from the previous ``docutils.writers.pep`` based system.
+Paragraphs are to be made compact where possible by omitting <p> tags, and
+footnote references should be enclosed in square brackets.
+
+
+3.7 Prepare for export to Jinja2
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Finally in ``pep_html_builder``, we gather all the parts to be passed to the
+Jinja2 template.
+This is also where we create the sidebar table of contents.
+
+The HTML files are then written out to the build directory.
+
+
+3. Theme
+--------
+
+The theme is comprised of the HTML template in
+``pep_sphinx_extensions/pep_theme/templates/page.html`` and the stylesheets in
+``pep_sphinx_extensions/pep_theme/static``.
+
+The template is entirely self-contained, not relying on any default behaviour
+from Sphinx.
+It specifies the CSS files to include, the favicon, and basic semantic
+information for document structure.
+
+The stylesheets are in two parts, with ``style.css`` defining the meat of the
+layout, and ``mq.css`` defining media-queries for a responsive layout.
+
+
+5. \PEP 0
+---------
+
+PEP 0 generation happens in three phases.
+Firstly, the text file itself is generated, it is added to Sphinx, and the data
+is post processed.
+
+
+5.1 File creation
+~~~~~~~~~~~~~~~~~
+
+``pep-0000.rst`` is created during a callback before documents are loaded by
+Sphinx.
+
+Firstly, we parse the individual PEP files, getting the
+RFC2822 header, and parsing and then validating that metadata.
+
+After collecting and validating all the PEP data, the creation of the index
+itself is in three steps:
+
+    1. Output header text.
+    2. Output the category and numerical indices
+    3. Output the author index
+
+The ``AUTHOR_OVERRIDES.csv`` file can be used to override an author's name in
+the PEP 0 output.
+
+We then add the newly created PEP 0 file to two Sphinx variables so that it will
+be processed as a normal file.
+
+
+5.2 Post processing
+~~~~~~~~~~~~~~~~~~~
+
+The ``PEPHeaders`` transform schedules the \PEP 0 post-processing code.
```

Comment:
```suggestion
The ``PEPHeaders`` transform schedules the PEP 0 post-processing code.
```

Another typo, or is there a reason? I don't see it written like this anywhere else...